packages/material-ui/src/styles/experimentalStyled.js

```diff
@@ -81,7 +83,8 @@ const experimentalStyled = (tag, options, muiOptions = {}) => {
   }
 
   const defaultStyledResolver = styled(tag, {
-    shouldForwardProp,
+    // ...((!componentSlot || componentSlot === 'Root') ? { shouldForwardProp } : { shouldForwardProp: slotShouldForwardProp }),
+    ...((!componentSlot || componentSlot === 'Root') && { shouldForwardProp }),
```

Comment:
I don't think we can fix this for all components, as some of them are build on top of primitive elements, others on components. For the ones build on top of primitive elements, sometimes they support `Component` prop that can change them to components. I think we will need to use something similar to https://github.com/mui-org/material-ui/blob/next/packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js#L642